mesonbuild/interpreter.py

```diff
@@ -4008,6 +4008,8 @@ def func_run_target(self, node, args, kwargs):
             if isinstance(i, dependencies.ExternalProgram) and not i.found():
                 raise InterpreterException(f'Tried to use non-existing executable {i.name!r}')
             cleaned_args.append(i)
+        if isinstance(cleaned_args[0], str):
+            cleaned_args[0] = self.func_find_program(node, cleaned_args[0], {})
```

Comment:
While at it, `add_*_script()` does some custom lookup in _find_source_script() instead of using find_program() that is subtly not the same thing. IMHO we should remove that code to use fin_program().

And run_command_impl() also does yet another subtly different thing.

Reply:
I'll file a separate MR for those.